packages/vscode-extension/src/utilities/packageManager.ts

```diff
@@ -32,6 +33,17 @@ export async function resolvePackageManager(): Promise<PackageManagerName> {
     } else if (isNpm) {
       return "npm";
     }
+    try {
+      const packageManager = require(path.join(workspacePath, "package.json")).packageManager;
```

Comment:
How often is this specified? Why we're doing regex matching on it? Don't we want to use lock-files for detection?

Reply:
We do use lock files to detect it but in rare case that they do not exist, we try checking the package.json instead of immediately fallbacking to npm. 
We perform regex to get rid of package managers version in the name e.g. `yarn@1.2.2` 

It is specified in most projects started with RN CLI